pkg/ruler/remotequerier_test.go

```diff
@@ -717,63 +715,6 @@ func TestRemoteQuerier_QueryReqTimeout(t *testing.T) {
 	require.Error(t, err)
 }
 
-func TestRemoteQuerier_BackoffRetry(t *testing.T) {
```

Comment:
Note, these tests duplicate what is already covered in `TestRemoteQuerier_QueryRetryOnFailure`. I don't think, testing the actual number of retries worth this number of code. The `backoff` is an internal implementation detail (_that's IMHO, and I don't feel strong about it).